Some 4X4 Resource Packs are attempting to format data outside of the image

```
Exception in thread "Thread-0" java.lang.RuntimeException: java.awt.image.RasterFormatException: (x + width) is outside of Raster
	at com.agentdid127.resourcepack.library.Util.propagate(Util.java:229)
	at com.agentdid127.resourcepack.forwards.ForwardsPackConverter.runPack(ForwardsPackConverter.java:123)
	at com.agentdid127.resourcepack.forwards.ForwardsPackConverter.lambda$runDir$0(ForwardsPackConverter.java:131)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1939)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at com.agentdid127.resourcepack.forwards.ForwardsPackConverter.runDir(ForwardsPackConverter.java:131)
	at com.agentdid127.resourcepack.GUI.lambda$null$0(GUI.java:55)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.awt.image.RasterFormatException: (x + width) is outside of Raster
	at java.desktop/sun.awt.image.ByteInterleavedRaster.createWritableChild(ByteInterleavedRaster.java:1241)
	at java.desktop/java.awt.image.BufferedImage.getSubimage(BufferedImage.java:1203)
	at com.agentdid127.resourcepack.library.utilities.ImageConverter.subImage2(ImageConverter.java:198)
	at com.agentdid127.resourcepack.library.utilities.ImageConverter.subImage(ImageConverter.java:100)
	at com.agentdid127.resourcepack.forwards.impl.textures.PaintingConverter.painting(PaintingConverter.java:83)
	at com.agentdid127.resourcepack.forwards.impl.textures.PaintingConverter.convert(PaintingConverter.java:35)
	at com.agentdid127.resourcepack.forwards.ForwardsPackConverter.runPack(ForwardsPackConverter.java:118)
	... 15 more
```

--- a/library/src/main/java/com/agentdid127/resourcepack/library/utilities/ImageConverter.java
+++ b/library/src/main/java/com/agentdid127/resourcepack/library/utilities/ImageConverter.java
@@ -8,10 +8,10 @@ import java.io.IOException;
 import java.nio.file.Path;
 
 public class ImageConverter {
-    protected int imageWidth = 0;
-    protected int imageHeight = 0;
-    protected int defaultW = 1;
-    protected int defaultH = 1;
+    protected int imageWidth;
+    protected int imageHeight;
+    protected int defaultW;
+    protected int defaultH;
     protected BufferedImage image;
     protected BufferedImage newImage;
     protected Path location;

--- a/library/src/main/java/com/agentdid127/resourcepack/library/utilities/ImageConverter.java
+++ b/library/src/main/java/com/agentdid127/resourcepack/library/utilities/ImageConverter.java
@@ -34,10 +34,6 @@ public class ImageConverter {
         imageHeight = image.getHeight();
 
         // check if widths and heights are legal.
-        if (imageWidth == 0 || imageHeight == 0) {
-            throw new IllegalStateException("Image must have a width or height greater than zero.");
-        }
-
         if (defaultW == 0 || defaultH == 0) {
             throw new IllegalStateException("Image must have a default width or height greater than zero.");
         }

--- a/library/src/main/java/com/agentdid127/resourcepack/library/utilities/ImageConverter.java
+++ b/library/src/main/java/com/agentdid127/resourcepack/library/utilities/ImageConverter.java
@@ -32,6 +32,17 @@ public class ImageConverter {
         defaultH = defaultHIn == null ? image.getHeight() : defaultHIn;
         imageWidth = image.getWidth();
         imageHeight = image.getHeight();
+
+        // check if widths and heights are legal.
+        if (imageWidth == 0 || imageHeight == 0) {
+            throw new IllegalStateException("Image must have a width or height greater than zero.");
+        }
+
+        if (defaultW == 0 || defaultH == 0) {
+            throw new IllegalStateException("Image must have a default width or height greater than zero.");
+        }
+
+
         if (!fileIsPowerOfTwo()) {
             Logger.log("Image (" + image.getWidth() + "x" + image.getHeight() + ") '" + locationIn.getFileName()
                     + "' resolution size is not a power of 2. Converting to be so.");
@@ -116,6 +127,10 @@ public class ImageConverter {
      * @param newHeight
      */
     public void newImage(int newWidth, int newHeight, int type) {
+        if (newWidth == 0 || newHeight == 0) {
+            throw new IllegalArgumentException("Width and height must be greater than zero!");
+        }
+
         newImage = new BufferedImage(scaleWidth(newWidth), scaleHeight(newHeight), type);
         g2d = (Graphics2D) newImage.getGraphics();
     }
@@ -466,10 +481,7 @@ public class ImageConverter {
      * @return double
      */
     public double getWidthMultiplier() {
-        double wMultiplier = (double) imageWidth / (double) defaultW;
-        // Make sure to not have 0 multiplier or cause issues!
-        wMultiplier = wMultiplier < 1 ? 1 : wMultiplier;
-        return wMultiplier;
+        return (double) imageWidth / (double) defaultW;
     }
 
     /**
@@ -478,10 +490,7 @@ public class ImageConverter {
      * @return double
      */
     public double getHeightMultiplier() {
-        double hMultiplier = (double) imageHeight / (double) defaultH;
-        // Make sure to not have 0 multiplier or cause issues!
-        hMultiplier = hMultiplier < 1 ? 1 : hMultiplier;
-        return hMultiplier;
+        return (double) imageHeight / (double) defaultH;
     }
 
     public int scaleX(int x) {

--- a/library/src/main/java/com/agentdid127/resourcepack/library/utilities/ImageConverter.java
+++ b/library/src/main/java/com/agentdid127/resourcepack/library/utilities/ImageConverter.java
@@ -38,7 +38,6 @@ public class ImageConverter {
             throw new IllegalStateException("Image must have a default width or height greater than zero.");
         }
 
-
         if (!fileIsPowerOfTwo()) {
             Logger.log("Image (" + image.getWidth() + "x" + image.getHeight() + ") '" + locationIn.getFileName()
                     + "' resolution size is not a power of 2. Converting to be so.");

